cmds/echo.c

```diff
@@ -48,11 +50,18 @@ static int cmd_echo(char *s)
 		return -EINVAL;
 	}
 
+	/* convert to lowercase */
+	p = args[0];
+	for (p = args[0]; *p; p++) {
+		if (isupper(*p))
+			*p |= 0x20;
+	}
+
 	/* Set echo */
-	if (hal_strcmp(args[0], "ON") == 0 || hal_strcmp(args[0], "on") == 0) {
+	if ((args[0][0] == 'o') && (args[0][1] == 'n') && (args[0][2] == '\0')) {
```

Comment:
I'm not delving into the `cmd_getArgs()` implementation, but is that ok that the length of the string `args[0]` is not checked here as in the `hal_strcmp()` case it is ? Here only the presence of `\0` at third (assumed) byte is checked, what if the size of `args[0]` < 3 ?

Reply:
e.g.
```
command line: echo o
args[0][0] == 'o'  true
args[0][1] == 'n'  false ( '\0' character )
```
and the whole condition is `false`

Reply:
Exactly, that's the expected answer.